core/contracts/types/entitlement_data.go

```diff
@@ -155,9 +141,27 @@ func MarshalEntitlement(
 			EntitlementType:   rawEntitlement.EntitlementType,
 			RuleEntitlementV2: &ruleData,
 		}, nil
-	} else {
-		return Entitlement{}, fmt.Errorf("invalid entitlement type '%s'", rawEntitlement.EntitlementType)
+	} else if rawEntitlement.EntitlementType == ModuleTypeUserEntitlement {
+		abiDef := `[{"name":"getAddresses","outputs":[{"type":"address[]","name":"out"}],"constant":true,"payable":false,"type":"function"}]`
```

Comment:
is`ModuleTypeUserEntitlement` "legacy"? There is an order of precedence here. If EntitlementV2 is absent, then check and evaluate UserEntitlement. Would be good to add a comment so that someone less familiar with the code won't accidentally move UserEntitlement before EntitlementV2.

Reply:
There's actually no order of precedence here - this could be handled just as well with a switch statement. I can make that change if you'd like.